### PR TITLE
[sensor] Add support for TemperatureSensor

### PR DESCRIPTION
**Pull request by jimmy-huang**
Refactor ARC code to support temperature sensors, also added
new sample BMI160Temperature.js that monitors the BMI160 sensor

Signed-off-by: Jimmy Huang <jimmy.huang@intel.com>

### CODE REVIEW

**Comment by grgustaf** [comment]
I guess add documentation, and try to clarify that this is of limited usefulness as it's the chip temperature. I'm not totally sure why it's there, unless to detect when it's too hot or cold to measure accurately? Mine shows like 84°F FWIW, quite a bit above room temperature.

Otherwise, LGTM.

**Comment by jimmy-huang** [comment]
Yeah, I have a separate patch that adds documentation, but I'll include it in this one and probably add some more

**Comment by grgustaf** [comment]
+1

File: docs/sensors.md [hunk 1]
```diff
@@ -12,11 +12,13 @@ ZJS Generic Sensor API implements the W3C Sensor API, and it's intended to
...
 accelerometer and gyroscope. Since the W3C Sensor API is still a draft spec,
 our implementation only provide a subset of the API and the API could be
-slightly different, but we try to follow the latest spec as closely as possible.
+slightly different, but we try to follow the latest spec as closely as
+possible.
 
-The currently supported hardware are Arduino 101s with its built-in BMI160
-accelerometer and gyroscope, also ambient light sensor that can be connected
-using analog pin.
+Note: The currently supported hardware are Arduino 101s with its built-in
```
Comment (grgustaf) [inline docs/sensors.md]: are Arduino 101s -> is Arduino 101

File: docs/sensors.md [hunk 2]
```diff
@@ -26,17 +28,18 @@ specific API functions.
...
-    readonly attribute SensorState state;   // The current state of Sensor object
-    attribute double frequency;             // The frequency set
-    void start();                           // Starts the sensor
-    void stop();                            // Stops the sensor
-    attribute ChangeCallback onchange;      // Callback handler for change events
-    attribute ActivateCallback onactivate;  // Callback handler for activate events
-    attribute ErrorCallback onerror;        // Callback handler for error events
+    readonly attribute SensorState state;   // current state of Sensor object
+    readonly attribute double timestamp;    // timestamp of the latest reading in milliseconds
+    attribute double frequency;             // frequency set
```
Comment (grgustaf) [inline docs/sensors.md]: Maybe "sampling frequency in hertz"?

Can you actually pass a float?

Comment (jimmy-huang) [inline docs/sensors.md]: not sure, but i dont see why not if your board can support frequency in float

**Pull request merged**